src/Trial.hs

```diff
@@ -142,33 +145,38 @@ instance Bifoldable Trial where
 
 instance Bitraversable Trial where
     bitraverse :: (Applicative f) => (e1 -> f e2) -> (a -> f b) -> Trial e1 a -> f (Trial e2 b)
-    bitraverse ef _ (Fiasco es)    = Fiasco <$> traverse (traverse ef) es
-    bitraverse ef ea (Result es a) = Result <$> traverse ef es <*> ea a
+    bitraverse ef _ (Fiasco es)    = Fiasco <$> traverseDList (traverse ef) es
+    bitraverse ef ea (Result es a) = Result <$> traverseDList ef es <*> ea a
     {-# INLINE bitraverse #-}
 
+{- 'DList' doesn't have a 'Traversable' instance -}
+traverseDList :: (Applicative f) => (a -> f b) -> DList a -> f (DList b)
+traverseDList f = fmap DL.fromList . traverse f . DL.toList
+{-# INLINE traverseDList #-}
```

Comment:
Very interesting! Would you like to open an issue in `DList` and possibly implement this feature? :slightly_smiling_face: